ingress/controllers/nginx-alpha-ssl/controller.go

```diff
@@ -0,0 +1,217 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"reflect"
+	"text/template"
+
+	vault "github.com/hashicorp/vault/api"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/util"
+)
+
+const (
+	nginxConf = `
+daemon off;
+
+events {
+	worker_connections 1024;
+}
+http {
+	# http://nginx.org/en/docs/http/ngx_http_core_module.html
+	types_hash_max_size 2048;
+	server_names_hash_max_size 512;
+	server_names_hash_bucket_size 64;
+
+	log_format default '$host $remote_addr - $remote_user [$time_local] \"$request\" $status $body_bytes_sent \"$http_referer\" \"$http_user_agent\"';
+
+{{range $ing := .Items}}
+{{range $rule := $ing.Spec.Rules}}
+	server {
+		server_name {{$rule.Host}};
+{{range $key, $val := $ing.Labels}}
+{{if (and (eq $key "ssl") (eq $val "true"))}}
+		listen 443 ssl;
+		ssl_certificate	/etc/nginx/certs/{{$rule.Host}}.crt;
+		ssl_certificate_key	/etc/nginx/certs/{{$rule.Host}}.key;
+{{- end}}{{end}}
+		error_log /dev/stdout info;
+		access_log /dev/stdout default;
+
+		listen 80;
+{{ range $path := $rule.HTTP.Paths }}
+		location {{$path.Path}} {
+			proxy_set_header Host $host;
+			proxy_pass http://{{$path.Backend.ServiceName}}.{{$ing.Namespace}}.svc.cluster.local:{{$path.Backend.ServicePort}};
+		}{{end}}
+	}{{end}}{{end}}
+}`
+)
+
+const nginxConfDir = "/etc/nginx"
+const nginxCommand = "nginx"
+
+// shellOut runs an external command.
+// stdout and stderr are attached to this external process
+func shellOut(shellCmd string, args []string) {
+	cmd := exec.Command(shellCmd, args...)
+	stdout, _ := cmd.StdoutPipe()
+	stderr, _ := cmd.StderrPipe()
+
+	fmt.Printf("Starting %v %v\n", shellCmd, args)
+
+	go io.Copy(os.Stdout, stdout)
+	go io.Copy(os.Stderr, stderr)
+	err := cmd.Start()
+	if err != nil {
+		log.Fatalf("Failed to execute %v: err: %v", cmd, err)
+	}
+}
+
+func main() {
+	var ingClient client.IngressInterface
+	if kubeClient, err := client.NewInCluster(); err != nil {
+		log.Fatalf("Failed to create client: %v.", err)
+	} else {
+		ingClient = kubeClient.Extensions().Ingress(api.NamespaceAll)
+	}
+	/* vaultEnabled
+	The following environment variables should be set:
+	VAULT_ADDR
+	VAULT_TOKEN
+	VAULT_SKIP_VERIFY (if using self-signed SSL on vault)
+	The only one we need to explicitly introduce is VAULT_ADDR, but we can check the others
+	*/
+	vaultEnabled := true
+	config := vault.DefaultConfig()
+	config.Address = os.Getenv("VAULT_ADDR")
+	vault, err := vault.NewClient(config)
+	if err != nil {
+		fmt.Printf("WARN: VAULT_ADDR is not set\n")
+		vaultEnabled = false
+	}
+
+	token := os.Getenv("VAULT_TOKEN")
```

Comment:
How about actually adding an RC+Service yaml for vault, and then polling here till the vault service is available (--wait-for-value or something can turn off the polling, but if it's true we just don't proceed until vault is up).